### PR TITLE
feat(unlock-app) - Settings page hooks

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

**Form validation** 
Screenshot 2022-11-21 at 13 02 50

Screenshot 2022-11-21 at 13 38 15

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
@kalidiagne I think it would be good to use a toggle above the field to let the user decide to not use a specific hook and in that case, use the `0x0` value for that one.

**Review by julien51 — CHANGES_REQUESTED** [review]
Looks good but for simplicity use a toggle above the field (similar to how we do `unlimnited` or `free`).

**Comment by kalidiagne** [comment]
> Looks good but for simplicity use a toggle above the field (similar to how we do `unlimnited` or `free`).
Screenshot 2022-11-23 at 13 23 42

Screenshot 2022-11-23 at 11 05 56

File: packages/unlock-js/src/PublicLock/v12/setEventHooks.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+export default async function (
+  {
+    lockAddress,
+    keyPurchase = '',
```
Comment (julien51) [inline packages/unlock-js/src/PublicLock/v12/setEventHooks.js:4]: The default value should be the Zero address

**Review by julien51 — APPROVED** [review]
LGTM!

**Pull request merged**